app/services/webhook/props.go

```diff
@@ -0,0 +1,161 @@
+package webhook
+
+import (
+	"context"
+	"fmt"
+	"github.com/getfider/fider/app"
+	"github.com/getfider/fider/app/models/dto"
+	"github.com/getfider/fider/app/models/entity"
+	"github.com/getfider/fider/app/models/enum"
+	"github.com/getfider/fider/app/pkg/web"
+	"time"
+)
+
+var dummyPost = &entity.Post{
+	ID:     36,
+	Number: 36,
+	Title:  "Example dummy post title",
+	Slug:   "example-dummy-post-title",
+	Description: `*Those data are fake.*
+
+This is an example of a post, but don't worry, nothing was created on your Fider instance:
+It was just a test of a webhook trigger.
+
+You can use ` + "`" + `{{ markdown .post_description }}` + "`" + ` to parse __Markdown__ to **HTML** in post!`,
+	CreatedAt: time.Date(2021, time.May, 7, 18, 42, 27, 0, time.UTC),
+	User: &entity.User{
+		ID:    7,
+		Name:  "Fider",
+		Email: "contact@fider.io",
+		Role:  1,
+	},
+	HasVoted:      true,
+	VotesCount:    7,
+	CommentsCount: 3,
+	Status:        enum.PostStarted,
+	Response: &entity.PostResponse{
+		Text:        "This is a response, still in *Markdown*.",
+		RespondedAt: time.Date(2021, time.July, 9, 15, 29, 57, 0, time.UTC),
+		User:        nil,
+	},
+	Tags: []string{"tag1", "tag2"},
+}
+
+func dummyTriggerProps(c context.Context, webhookType enum.WebhookType) dto.Props {
+	props := dto.Props{}
+	author := c.Value(app.UserCtxKey).(*entity.User)
+	tenant := c.Value(app.TenantCtxKey).(*entity.Tenant)
+	baseURL, logoURL := web.BaseURL(c), web.LogoURL(c)
+	dummyPost.User.AvatarURL = logoURL
+	dummyPost.Response.User = author
+	DescribeUser(props, author, "author")
+	DescribeTenant(props, tenant, "tenant", baseURL, logoURL)
+	switch webhookType {
+	case enum.WebhookNewPost:
+		DescribePost(props, dummyPost, "post", baseURL, false, false)
+	case enum.WebhookNewComment:
+		DescribePost(props, dummyPost, "post", baseURL, true, true)
+		props["comment"] = "An example **comment** on a post."
+	case enum.WebhookChangeStatus:
+		DescribePost(props, dummyPost, "post", baseURL, true, true)
+		props["post_old_status"] = enum.PostOpen
+	case enum.WebhookDeletePost:
+		DescribePost(props, dummyPost, "post", baseURL, true, true)
+		props["post_status"] = enum.PostDeleted
+		props["post_response_text"] = "The reason _why_ this post was deleted."
+	}
+	return props
+}
+
+func DescribeUser(props dto.Props, user *entity.User, keyPrefix string) {
+	if user == nil {
+		return
+	}
+	role, _ := user.Role.MarshalText()
```

Comment:
most enums should have a String() function (which comes from the Stringer interface), have a look at the AvatarType enum, doing that means we can inline this on the function below

Reply:
Interesting, I didn't noticed it was on the `AvatarType` enum. I'll implement it on the `Role` enum as well so :)